gateway/api/schedule.py

```diff
@@ -13,26 +14,60 @@
 
 from opentelemetry import trace
 
-from api.models import Job, Program, ComputeResource
+from api.models import Job, Program, ComputeResource, ProgramConfig
 from api.ray import submit_job, create_ray_cluster, kill_ray_cluster
 from main import settings as config
 
 User: Model = get_user_model()
 logger = logging.getLogger("commands")
 
 
-def save_program(serializer, request) -> Program:
+def save_program(request) -> Program:
     """Save program.
 
     Args:
-        serializer: program serializer with data attached.
+        request: request data.
 
     Returns:
         saved program
     """
-    program = Program(**serializer.data)
+    if request.data.get("config"):
```

Comment:
model serializer should be able to handle this. I do not think you need any of this. 
You already implemented serializer, which is doing mapping between request data and Program object and all related object (which is config)

Reply:
I tried but I couldn't make it work with `models.ForeignKey` field.  I'll try it again.

Reply:
this might help https://www.django-rest-framework.org/api-guide/relations/#writable-nested-serializers